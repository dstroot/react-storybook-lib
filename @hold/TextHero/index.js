import React from "react";
import PropTypes from "prop-types";

const TextHero = ({ heading, text, link }) => {
  const textBig = {
    fontSize: `calc(26px + 1.5vw + 0.5vmin)`,
    lineHeight: "1.2",
    fontWeight: "300"
  };

  return (
    <div className="container mt-5 mb-5">
      <div className="row justify-content-center">
        <div className="col">
          <div className="text-center mx-auto">
            <p style={textBig}>{heading || ""}</p>
            <p className="lead">{text || ""}</p>
            <a href={link} target="_self" className="btn btn-primary">
              We can help!
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

TextHero.propTypes = {
  heading: PropTypes.string,
  text: PropTypes.string,
  link: PropTypes.string
};

export default TextHero;
