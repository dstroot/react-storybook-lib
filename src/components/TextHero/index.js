import React from 'react';
import PropTypes from 'prop-types';

const TextHero = ({ heading, text, link }) => {
  return (
    <div className="container-fluid mt-5 mb-5">
      <div className="row justify-content-center">
        <div className="col-8">
          <div className="text-center mx-auto">
            <h1 className="display-4">{heading || ''}</h1>
            <p className="lead">{text || ''}</p>
            <a href={link} target="_self" className="btn btn-primary">
              WE CAN HELP
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

TextHero.propTypes = {
  heading: PropTypes.string,
  text: PropTypes.string,
  link: PropTypes.string,
};

export default TextHero;
