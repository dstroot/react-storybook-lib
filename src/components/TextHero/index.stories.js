import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import TextHero from '../TextHero';

// TextHero section
storiesOf('TextHero', module).add('show TextHero', () => (
  <TextHero
    heading="This is a Heading"
    text="This is some very important marketing text..."
    link="https://somelink"
  />
));
