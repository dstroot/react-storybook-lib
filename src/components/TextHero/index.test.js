import React from 'react';
import renderer from 'react-test-renderer';
import TextHero from '../TextHero';

describe('TextHero', () => {
  it('it should render', () => {
    const component = renderer.create(<TextHero />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
