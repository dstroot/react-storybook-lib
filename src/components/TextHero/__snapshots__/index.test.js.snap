// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextHero it should render 1`] = `
<div
  className="container mt-5 mb-5"
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col"
    >
      <div
        className="text-center mx-auto"
      >
        <p
          style={
            Object {
              "fontSize": "calc(26px + 1.5vw + 0.5vmin)",
              "fontWeight": "300",
              "lineHeight": "1.2",
            }
          }
        >
          
        </p>
        <p
          className="lead"
        >
          
        </p>
        <a
          className="btn btn-primary"
          target="_self"
        >
          We can help!
        </a>
      </div>
    </div>
  </div>
</div>
`;
